queue/fixed_serializable.go

```diff
@@ -0,0 +1,674 @@
+	jobCopy, err := q.copyJob(j)
+	if err != nil {
+		grip.Error(message.Fields{
+			"message":  "could not copy job",
+			"job_id":   j.ID(),
+			"queue_id": q.ID(),
+		})
+		q.storage[name] = j
+		return
+	}
+	q.storage[name] = jobCopy
+}
+
+// getCopy finds and returns a copy of the job in storage. If it cannot create a
+// copy, it will return the job directly from storage.
+func (q *limitedSizeSerializableLocal) getCopy(name string) (amboy.Job, bool) {
+	j, ok := q.storage[name]
+	if !ok {
+		return nil, false
+	}
+	jobCopy, err := q.copyJob(j)
+	if err != nil {
+		grip.Debug(message.WrapError(err, message.Fields{
+			"message":  "could not copy job",
+			"job_id":   j.ID(),
+			"queue_id": q.ID(),
+		}))
+		return j, true
+	}
+	return jobCopy, true
+}
+
+// jobStored returns whether or not a job is in storage.
+func (q *limitedSizeSerializableLocal) jobStored(name string) bool {
+	_, ok := q.storage[name]
+	return ok
+}
+
+// Start starts the runner and initializes the pending job storage. Only
+// produces an error if the underlying runner fails to start.
+func (q *limitedSizeSerializableLocal) Start(ctx context.Context) error {
+	q.mu.Lock()
+	defer q.mu.Unlock()
+
+	if q.started {
+		return errors.New("cannot start a running queue")
+	}
+
+	q.lifetimeCtx = ctx
+
+	err := q.runner.Start(ctx)
+	if err != nil {
+		return err
+	}
+
+	if q.retryHandler != nil {
+		if err := q.retryHandler.Start(ctx); err != nil {
+			return errors.Wrap(err, "starting retry handler")
+		}
+		go q.monitorStaleRetryingJobs(ctx)
+	}
+
+	q.started = true
+
+	grip.Info("job server running")
+
+	return nil
+}
+
+func (q *limitedSizeSerializableLocal) monitorStaleRetryingJobs(ctx context.Context) {
+	defer func() {
+		if err := recovery.HandlePanicWithError(recover(), nil, "stale retry job monitor"); err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":  "stale retry job monitor failed",
+				"queue_id": q.ID(),
+			}))
+			go q.monitorStaleRetryingJobs(ctx)
+		}
+	}()
+
+	monitorInterval := defaultStaleRetryingMonitorInterval
+	if q.staleRetryingMonitorInterval != 0 {
+		monitorInterval = q.staleRetryingMonitorInterval
+	}
+	timer := time.NewTimer(0)
+	defer timer.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-timer.C:
+			q.handleStaleRetryingJobs(ctx)
+			timer.Reset(monitorInterval)
+		}
+	}
+}
+
+func (q *limitedSizeSerializableLocal) handleStaleRetryingJobs(ctx context.Context) {
+	q.mu.RLock()
+	defer q.mu.RUnlock()
+	for _, j := range q.storage {
+		if !j.RetryInfo().ShouldRetry() {
+			continue
+		}
+		if j.RetryInfo().ShouldRetry() && time.Since(j.Status().ModificationTime) < q.info().LockTimeout {
+			continue
+		}
+		grip.Error(message.WrapError(q.retryHandler.Put(ctx, j), message.Fields{
+			"message":  "could not enqueue stale retrying job",
+			"job_id":   j.ID(),
+			"queue_id": q.ID(),
+		}))
+	}
+}
+
+func (q *limitedSizeSerializableLocal) Close(ctx context.Context) {
```

Comment:
None of the other implementations do it. I think the assumption is that once it's closed, closing it again won't do anything anyways since nothing is running.